src/runtime/recipe/slot.ts

```diff
@@ -98,6 +98,10 @@ export class Slot {
     return 0;
   }
 
+  findHandleByID(id) {
+    return this.handleConnections.find(handleConn => handleConn.handle && handleConn.handle.id === id).handle;
```

Comment:
Won't this cause an error if no handle is found with the requested id? This would be safer:
```
const connection = this.handleConnections.find(...);
return connection && connection.handle;
```

(I realise this was just how the existing code had it, but we might as well guard against bugs when we can :slightly_smiling_face:)